monitor_correct_errors/models/monitor_correct_errors.py

```diff
@@ -0,0 +1,336 @@
+            ('sent_date', '=', False),
+            ('letter_delivered', '=', False)])
+
+        date_now = fields.Date.from_string(fields.Date.today())
+
+        list_error_correspondence = list_correspondence.filtered(
+            lambda l: (date_now - fields.Date.from_string(
+                l.status_date)).days > 30)
+
+        error_log = {}
+        error_log['error_type'] = "B2S letters not sent"
+
+        for correspondence in list_error_correspondence:
+            error_log['record'] = "Letters not send"
+            error_log['record_id'] = correspondence.id
+            error_log['record_name'] = correspondence.name
+            error_log['record_model'] = "correspondence"
+
+            self.error_log_list.append(self.env['error_log'].create(error_log))
+
+        list_correspondence_translation_queue = self.env['correspondence'].\
+            search([('state', '=', 'Global Partner translation queue')])
+
+        list_correspondence_translation_queue_error = \
+            list_correspondence_translation_queue.filtered(
+                lambda l: (date_now - fields.Date.from_string(
+                    l.status_date)).days > 30)
+
+        error_log['error_type'] = "Letters in translation queue for too long"
+        for correspondence in list_correspondence_translation_queue_error:
+            error_log['record'] = "Letters in queue for too long"
+            error_log['record_id'] = correspondence.id
+            error_log['record_name'] = correspondence.name
+            error_log['record_model'] = "correspondence"
+
+            self.error_log_list.append(self.env['error_log'].create(error_log))
+
+        list_correspondence_scan = self.env['correspondence'].search([
+            ('state', '=', 'Received in the system')])
+
+        list_correspondence_scan_error = list_correspondence_scan.filtered(
+            lambda l: (date_now - fields.Date.from_string(
+                l.status_date)).days > 30)
+
+        error_log['error_type'] = "Letters scanned in for too long"
+
+        for correspondence in list_correspondence_scan_error:
+            error_log['record'] = "Letters scanned in for too long"
+            error_log['record_id'] = correspondence.id
+            error_log['record_name'] = correspondence.name
+            error_log['record_model'] = "correspondence"
+
+            self.error_log_list.append(self.env['error_log'].create(error_log))
+
+
+    def _fix_sponsorship(self):
+        '''
+        This method fix the sponsorships that are not in the correct state, or
+        sponsorship where the child has no link to the sponsor.
+        :return:
+        '''
+
+        sponsorships = self.env['recurring.contract'].search([
+            ('state', 'not in', ['terminated', 'cancelled']),
+            ('type', '=', 'S')])
+
+        error_log = {}
+        error_log['error_type'] = "Sponsored child not in correct state"
+
+        child_wrong_state = sponsorships.filtered(
```

Comment:
filtered is much slower than search method. Why not directly do the search here with all criterias? 

```
child_wront_state = self.env['recurring.contract'].search([
             ('state', '=', 'active'),
             ('type', '=', 'S'),
             ('child_id.sponsor_id', '!=', False),
             ('child_id.state', '!=', 'P')
]).mapped('child_id')
```